Mathlib/Order/CompleteLattice.lean

```diff
@@ -1293,6 +1293,27 @@ theorem iInf_extend_top {e : ι → β} (he : Injective e) (f : ι → α) :
     ⨅ j, extend e f ⊤ j = iInf f :=
   @iSup_extend_bot αᵒᵈ _ _ _ _ he _
 
+section le
+
+variable {ι : Type*} [PartialOrder ι] (f : ι → α) (i : ι)
+
+theorem biSup_le_eq_sup : (⨆ j ≤ i, f j) = (⨆ j < i, f j) ⊔ f i := by
+  rw [iSup_split_single _ i]
+  simp only [le_refl, iSup_pos, iSup_and', lt_iff_le_and_ne, and_comm, sup_comm]
```

Comment:
Does squeezing this simp make a performance difference? If not, can you unsqueeze this, please?

Reply:
For both there's a ~10x slowdown from unsqueezing, but still plenty fast in either case (~420 vs ~4200 heartbeats). I think I'll leave them squeezed, unless you feel strongly.